src/main/java/io/dropwizard/sharding/dao/RelationalDao.java

```diff
@@ -241,6 +313,43 @@ public boolean update(String parentKey, DetachedCriteria criteria, Function<T, T
         }
     }
 
+    <U> boolean selectAndUpdateOrSave(LookupDao.LockedContext<U> context, DetachedCriteria criteria, Function<T, T> updater) {
```

Comment:
This will cause issues somewhere in client code. The `updater` here is also the creator. This is problematic.

Reply:
Figure out a cleaner way for doing this.